cdci_data_analysis/analysis/instrument.py

```diff
@@ -166,6 +166,9 @@ def run_query(self, product_type,
         if logger is None:
             logger = self.get_logger()
 
+        #  this was removed by 2f5b5dfb7e but turns out it is used by some plugins, see test_server_plugin_integral_all_sky
+        self._current_par_dic=par_dic
```

Comment:
Which case brakes?
Because as far as I  remember, _current_par_dic is no longer used anywhere

Reply:
it says it in the comment: this very case here test_server_plugin_integral_all_sky

Reply:
here: 

https://github.com/oda-hub/dispatcher-plugin-integral-all-sky/blob/master/dispatcher_plugin_integral_all_sky/spiacs_lightcurve_query.py#L311

maybe this breaks encapsulation, it needs to be determined if it is actually necessary to support this interface to plugins.

Reply:
It is used for the definition of a product, but honestly I don't know the details why:
[https://github.com/oda-hub/dispatcher-app/blob/6330adeb99832ec23d05943e705badb87badd10a/cdci_data_analysis/analysis/products.py#L327-L373](https://github.com/oda-hub/dispatcher-app/blob/6330adeb99832ec23d05943e705badb87badd10a/cdci_data_analysis/analysis/products.py#L327-L373)

Reply:
> It is used for the definition of a product, but honestly I don't know the details why:

oh it is used to add url to fits files. This url is a unique reference to the product, and it can be resolved to actually get the product from the frontend. 

It is essential that these urls are built and maintained. It is also the url I mentioned in [there](https://github.com/oda-hub/dispatcher-app/issues/20#issuecomment-818571887).

What is unclear, however, is why should this possibly be done customly in  this plugin, by accessing presumably internal (_.*) attribute. All plugins should do this equally, and hence it can be done in the dispatcher base class.

But as of now, looking in other plugins, the same approach is used everywhere: 
(e.g. [polar](https://github.com/oda-hub/dispatcher-plugin-polar/blob/master/cdci_polar_plugin/polar_lightcurve_query.py#L243))

But, before we do any refactoring, this functionality should be maintained.

This PR here also serves as an example of the kind of near-production cases we need before continious deployment can be done safely.